src/resolver/resolver.zig

```diff
@@ -2846,6 +2846,12 @@ pub const Resolver = struct {
                 const total_length: ?u32 = strings.indexOfChar(original_path, '*');
                 var prefix_parts = [_]string{ abs_base_url, original_path[0 .. total_length orelse original_path.len] };
 
+                // Concatenate the matched text with the suffix from the wildcard path
+                const suffix = std.mem.trimLeft(u8, original_path[total_length orelse original_path.len ..], "*");
+                var matched_text_with_suffix = r.allocator.alloc(u8, matched_text.len + suffix.len) catch unreachable;
```

Comment:
1) Please do this without dynamically allocating temporary memory, possibly adding another temporary buffer to the threadlocal list used elsewhere in this file

2) Do we need to do this on each call? Can we avoid doing it in certain cases?

Reply:
Since the threadlocal buffers are a fixed size, wouldn't using one of them mean that when it's joining the other strings together at the end, it would be trying to join a buffer that is larger than its contents and try to access indexes that don't exist?

Good point, this only needs to happen when `total_length != null` since that is the only time this value is consumed.

Reply:
the size of the buffers we use are the max allowed filepath length. we could add 2x to that. but filesystems by default limit the path length to be around 1024 bytes (4096 if linux)

Reply:
I just pushed a new version using a buffer in the threadlocal list. Is this more like what you had in mind?